.github/workflows/flags.yml

```diff
@@ -57,20 +61,19 @@ jobs:
           os:  ${{ runner.os }}
 
       - name: Build `hls-graph` with flags
-        run: cabal v2-build hls-graph --flags="pedantic embed-files stm-stats"
+        run: cabal v2-build hls-graph --flags="embed-files stm-stats"
 
       - name: Build `hie-compat` with flags
         run: cabal v2-build hie-compat --flags="ghc-lib"
 
-      - name: Build `hls-plugin-api` with flags
-        run: cabal v2-build hls-plugin-api --flags="pedantic"
-
-      - name: Build `hls-test-utils` with flags
-        run: cabal v2-build hls-test-utils --flags="pedantic"
-
-      - name: Build
+      - name: Build `ghcide` with flags
         run: cabal v2-build ghcide --flags="ghc-patched-unboxed-bytecode test-exe executable bench-exe"
 
+      # we have to clean up warnings for 9.0 and 9.2 before enable -WAll
+      - if: matrix.ghc != '9.0.2' && matrix.ghc != '9.2.1'
+        name: Build with pedantic (-WError)
+        run: cabal v2-build --flags="pedantic"
```

Comment:
This will rebuild everything, I think? do we want that? seems like it'll make the workflow slow. Maybe it's okay.

Reply:
only local packages (as `--flags` only affect them), even the ones without the flag as we have one package in the root of the local dependency tree with the flag available